Detectors/TPC/workflow/src/TPCTimeSeriesSpec.cxx

```diff
@@ -952,12 +978,15 @@ class TPCTimeSeries : public Task
       const auto dedxQTotVars = getdEdxVars(0, track);
       const auto dedxQMaxVars = getdEdxVars(1, track);
 
-      const int nClITS = hasITSTPC ? tracksITS[tracksITSTPC[idxITSTPC.front()].getRefITS().getIndex()].getNClusters() : -1;
-      float chi2ITS = hasITSTPC ? tracksITS[tracksITSTPC[idxITSTPC.front()].getRefITS().getIndex()].getChi2() : -1;
+      // make check to avoid crash in case no or less ITS tracks have been found!
+      const int idxITSTrack = hasITSTPC ? tracksITSTPC[idxITSTPC.front()].getRefITS().getIndex() : -1;
+      const bool idxITSCheck = (idxITSTrack < tracksITS.size()) && hasITSTPC;
```

Comment:
Please check that the refITS.getSource() returns 0, otherwise this is not an ITS track but afterburner tracklet reference to different container

Reply:
Thanks, this explains the bug with the index! I just implemented the check